gh-134209: use heap-allocated memory in `_curses.window.{instr,getstr}` (#134283)
python/cpython: Doc/library/curses.rst

            window.getstr(y, x, n)

   Read a bytes object from the user, with primitive line editing capacity.

>    The maximum of *n* for this method is 2047.

Reviewer: ```suggestion
   The maximum value for *n* is 2047.
```

Sorry, I missed this phrasing.

>    .. versionchanged:: next
>       The maximum value for *n* is set to 2047.

Reviewer: ```suggestion
   .. versionchanged:: next
      The maximum value for *n* is set to 2047.

```

Sorry, we actually separate functions by two blank lines... Thanks for your patience by the way.

            window.getstr(y, x, n)

   Read a bytes object from the user, with primitive line editing capacity.

>    The maximum of *n* for this method is 2047.

Reviewer: ```suggestion
   The maximum of *n* for this method is 2047.

   .. versionchanged:: next
      The maximum value for *n* is set to 2047.
```

   Return a bytes object of characters, extracted from the window starting at the
   current cursor position, or at *y*, *x* if specified. Attributes are stripped
   from the characters.  If *n* is specified, :meth:`instr` returns a string

>    at most *n* characters long (exclusive of the trailing NUL). The maximum of *n*
>    for this method is 2047.

Reviewer: ```suggestion
   at most *n* characters long (exclusive of the trailing NUL). The maximum of *n*
   for this method is 2047.

   .. versionchanged:: next
      The maximum value for *n* is set to 2047.
```

   Return a bytes object of characters, extracted from the window starting at the
   current cursor position, or at *y*, *x* if specified. Attributes are stripped
   from the characters.  If *n* is specified, :meth:`instr` returns a string

>    at most *n* characters long (exclusive of the trailing NUL). The maximum of *n*
>    for this method is 2047.

Reviewer: ```suggestion
   at most *n* characters long (exclusive of the trailing NUL).
   The maximum value for *n* is 2047.
```
